app/screens/sso/sso_with_redirect_url.tsx

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {useEffect, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {Linking, Platform, Text, TouchableOpacity, View} from 'react-native';
+import DeviceInfo from 'react-native-device-info';
+import {SafeAreaView} from 'react-native-safe-area-context';
+import urlParse from 'url-parse';
+
+import FormattedText from '@components/formatted_text';
+import Loading from '@components/loading';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {setDeepLinkUrl} from '@requests/local/systems';
+import {tryOpenURL} from '@utils/url';
+
+interface SSOWithRedirectURLProps {
+    loginError: string;
+    loginUrl: string;
+    onCSRFToken: (token: string) => void;
+    onMMToken: (token: string) => void;
+    setLoginError: (value: string) => void;
+    theme: Partial<Theme>
+}
+
+const SSOWithRedirectURL = ({loginError, loginUrl, onCSRFToken, onMMToken, setLoginError, theme}: SSOWithRedirectURLProps) => {
+    const [error, setError] = useState<string>('');
+    const style = getStyleSheet(theme);
+    const intl = useIntl();
+    let customUrlScheme = 'mmauth://';
+    if (DeviceInfo.getBundleId && DeviceInfo.getBundleId().includes('rnbeta')) {
+        customUrlScheme = 'mmauthbeta://';
+    }
+
+    const redirectUrl = customUrlScheme + 'callback';
+    const init = (resetErrors?: boolean) => {
+        if (resetErrors !== false) {
+            setError('');
+            setLoginError('');
+        }
+        const parsedUrl = urlParse(loginUrl, true);
+        parsedUrl.set('query', {
+            ...parsedUrl.query,
+            redirect_to: redirectUrl,
+        });
+        const url = parsedUrl.toString();
+
+        const onError = (e: Error) => {
+            let message;
+            if (e && Platform.OS === 'android' && e?.message?.match(/no activity found to handle intent/i)) {
+                message = intl.formatMessage({
+                    id: 'mobile.oauth.failed_to_open_link_no_browser',
+                    defaultMessage: 'The link failed to open. Please verify if a browser is installed in the current space.',
+                });
+            } else {
+                message = intl.formatMessage({
+                    id: 'mobile.oauth.failed_to_open_link',
+                    defaultMessage: 'The link failed to open. Please try again.',
+                });
+            }
+            setError(
+                message,
+            );
+        };
+        tryOpenURL(url, onError);
+    };
+
+    useEffect(() => {
+        const onURLChange = ({url}: { url: string }) => {
+            if (url && url.startsWith(redirectUrl)) {
+                // save deepLinkUrl under Global
+                setDeepLinkUrl('');
```

Comment:
do we need this?

Reply:
Yeah in our case this wouldn't be needed.. in fact we probably want the redirection to happen